packages/govuk-frontend/src/govuk/components/character-count/character-count.jsdom.test.mjs

```diff
@@ -1,28 +1,32 @@
+import { getExamples, renderComponent } from '@govuk-frontend/lib/components'
+
 import { CharacterCount } from './character-count.mjs'
 
 describe('CharacterCount', () => {
-  let $container
-  let $textarea
-
-  beforeAll(() => {
-    $container = document.createElement('div')
-    $textarea = document.createElement('textarea')
-
-    // Component checks that GOV.UK Frontend is supported
-    document.body.classList.add('govuk-frontend-supported')
+  let html
+
+  // Reset to a clean component within a clean `<body>` tag
+  beforeEach(async () => {
+    if (!html) {
+      const examples = await getExamples('character-count')
+      html = renderComponent(
+        'character-count',
+        examples['to configure in JavaScript']
+      )
+    }
 
-    // Component checks that required elements are present
-    $textarea.classList.add('govuk-js-character-count')
-    $container.appendChild($textarea)
+    document.documentElement.innerHTML = `
+      <body class="govuk-frontend-supported">${html}</body>
+    `
```

Comment:
Ah was it too risky using `document.body.innerHTML = html`?

Reply:
No, it's done to reset the `<body>` tag to a clean state after a test adds a `lang` attribute to it. I've added a comment so we know why it's that way in the future 😊

Reply:
Yeah that makes sense. Shall we keep `outerHTML` and `document.body` for familiarity?

```suggestion
    document.body.outerHTML = `<body class="govuk-frontend-supported">${html}</body>`
```

Reply:
Cheers. If you want to keep the new lines (but without trailing spaces) wrap the HTML in `outdent`?

Reply:
Not fussy about the whitespace, I just had edited the left side of the `=`.